test/service/benches/validate_block.rs

```diff
@@ -0,0 +1,164 @@
+// This file is part of Cumulus.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use codec::{Decode, Encode};
+use core::time::Duration;
+use criterion::{criterion_group, criterion_main, BatchSize, Criterion, Throughput};
+use cumulus_primitives_core::{relay_chain::AccountId, PersistedValidationData, ValidationParams};
+use cumulus_test_client::{
+	generate_extrinsic_with_pair, BuildParachainBlockData, InitBlockBuilder, TestClientBuilder,
+	ValidationResult,
+};
+use cumulus_test_relay_sproof_builder::RelayStateSproofBuilder;
+use cumulus_test_runtime::{BalancesCall, Block, Header, UncheckedExtrinsic};
+use cumulus_test_service::bench_utils as utils;
+use polkadot_primitives::HeadData;
+use sc_block_builder::BlockBuilderProvider;
+use sc_client_api::UsageProvider;
+use sc_executor_common::wasm_runtime::WasmModule;
+
+use sp_blockchain::{ApplyExtrinsicFailed::Validity, Error::ApplyExtrinsicFailed};
+
+use sp_core::{sr25519, Pair};
+
+use sp_runtime::{
+	traits::Header as HeaderT,
+	transaction_validity::{InvalidTransaction, TransactionValidityError},
+};
+
+fn create_extrinsics(
```

Comment:
Yeah this is a bit of a compromise. I am using the test client for the validate block benchmarks and test service for block production and block import since I wanted to reuse the nice convenience features of the test client block builder.